ui/src/components/user-details.tsx

```diff
@@ -0,0 +1,270 @@
+import { Component } from 'inferno';
```

Comment:
I spose this is fine, but I'm gonna have to test it a lot.

Reply:
I think I mightve over-thought this and I'll check how it works putting this back.  IMO kinda nice to split up some of the logic but with how the User component works this may not be the best way and if it isnt required for to fix the dupe'd requests I'm fine undoing it